### DISCUSSION TITLE
Date format in Birt 4.9 is coming out like this "20 de set. de 2022" in reports

### DISCUSSION THREAD

**Discussion by MuriloIPM** [eclipse-birt/birt#1045] [Q&A] [answered]
I have a reporting servlet with the BIRT Runtime, which was migrated from version 4.4 to version 4.9, but when issuing a report that has the date formatted as "Unformatted" I had a problem with the date formats, for example:

Version 4.4 -> 09/20/2022

Now it comes out like this (4.9) -> 20 de set. de 2022

It would be unfeasible for me to change the .rptdesign as it would be a very large amount, and making this change to the design at runtime would require more resources.

Would you have any tips on what to do or some configuration I can do in my EngineConfig something like that.

Java version: 7 -> **11(current)**

Tomcat version: 7 -> **9(current)**

My location: pt_BR

**Comment by hvbtup** [comment]
I don't think this is a BIRT specific issue.
Probably it's just that Java determines the locale in a different way.
Maybe this has been changed between the Java releases or (if you have a new environment) your environment ist different (e.g. the OS uses a different locale or the environment variables are different).

You should either set up Java to explicitly use a specific locale (see https://www.oracle.com/technical-resources/articles/javase/locale.html) by setting the `user.language` and `user.country` system properties,

or specify the locale inside the report itself for each and every data item containing a date.
Perhaps this can also be set globally once per report, I don't know.

**Reply by MuriloIPM** [reply]
I already checked that too, I checked that the default library java.text.DateFormat, used by birt, is that new format, if I give a Locale.getDefault() it returns the right locale.
I ran DateFormat.getDateTimeInstance() returns "d 'de' MMM 'y' HH:mm:ss".

**Comment by speckyspooky** [comment]
Hello MuriloIPM,
I started some tests with BIRT 4.8 and 4.9 (JVM11).
Currently I'm not sure in wich constellation you use your date (it seems for me a colum from dataset).
I tested parameters, dynamic text and date column without format and on my side I get te correct DE-format (DE ist default on my side).

Test result:
![datetest](https://user-images.githubusercontent.com/41593722/190842630-a8fea2a1-333e-49aa-bebb-bbb0a23f2a27.png)

Attached I added my small test cases based on classic models (data set).

Test file:
[date_test.zip](https://github.com/eclipse/birt/files/9591031/date_test.zip)

Can you test in on your side for the results. Additional on it if you haven't the option to change the format on rptdesign then the other option would be to try the date format on SQL-level.
(But it could be that for correct display that you must change the column data type from date to string to get the correct date format based on SQL because BIRT could be change again the format.)

Best regards

**Reply by MuriloIPM** [reply]
I understand, but I need to find some way to do this without having to change the SQL level or rptdesign because the amount of report that has been developed is very, very large.
I did the same test using a script that returns the current date, in both versions:
Birt 4.9 Tests:
[Birt_4.9.zip](https://github.com/eclipse/birt/files/9598546/Birt_4.9.zip)

Birt 4.4 Tests:
[Birt_4.4.zip](https://github.com/eclipse/birt/files/9598613/Birt_4.4.zip)

**Comment by hvbtup** [comment]
I still cannot imagine this is a BIRT issue.
Can you try to let your report show some import Java system properties in the output?
- In particular, `user.language` and `user.country`?
- And the result of Java's `Locale.getDefault()`?

**Reply by MuriloIPM** [reply]
`Locale.getDefault()`: 
- baseLocale
    - language=pt, region:br
    - hashCodeValue: 106956076
    - languageTag:"pt-BR"
    - LocaleExtenxions: null
---
- `-Duser.language`=pt
- `-Duser.country`=BR
---
Birt 4.9 Tests:
[Birt_4.9.zip](https://github.com/eclipse/birt/files/9598546/Birt_4.9.zip)

Birt 4.4 Tests:
[Birt_4.4.zip](https://github.com/eclipse/birt/files/9598613/Birt_4.4.zip)

**Comment by hvbtup** [comment]
I still think this is JVM issue, not a BIRT issue.
I reckon that the default date format has changed between the Java releases.

                        <structure name="dateTimeFormat">
                            <property name="category">Unformatted</property>
                        </structure>

You probably don't want to hear it, but  if you do not explicitly specify a date format in BIRT, then it means "I don't care".

Anyway, remember that BIRT reports (and libraries) are just XML files.

So you should be able to  fix the format for one of the reports, then write a little script that replaces this XML fragment

                            <property name="category">Unformatted</property>

with the corresponding XML fragment of the fixed report.

**Reply by MuriloIPM** [reply]
Changing the XML would be my option, so I found it strange comparing the java.text.DateFormat of java 7 and 11, from what I saw Birt uses this class, they are almost the same, but thanks for your attention. If you have another idea, just send it to me.

**Comment by hvbtup** [comment, marked as answer]
See also #1107
